### PR TITLE
Port joy node to ros2 (linux only)

### PR DESCRIPTION
**Pull request by mikaelarguedas**
I'll squash together most commits except the file rename so that we keep history of the file and dont struggle to much to integrate new change.
I'm going to open a PR upstream for porting the package.xml to format2 and another one for whitespaces.

Not sure what to do with version number and changelogs, looking for feedback on this.

Ready for review

replaces https://github.com/ros2/joystick_drivers_from_scratch/pull/2

CI:
turtlebot amd64: [Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux/36/)
turtlebot aarch64: [Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux-aarch64/50/)

### CODE REVIEW

File: joy/CMakeLists.txt
Comment (clalancette) [inline joy/CMakeLists.txt]: It seems to me that if we don't find joystick.h, the package should fail to build.  After all, the whole point of the package is to build the joy node.

File: joy/CMakeLists.txt
Comment (clalancette) [inline joy/CMakeLists.txt:37]: Do we need to do ament_export_{include_directories, libraries} here?

File: joy/src/joy_node_linux.cpp
Comment (clalancette) [inline joy/src/joy_node_linux.cpp:116]: Ah, I didn't know about `get_parameter_or`.  Much nicer than what I did :).

**Review by clalancette — COMMENTED** [review]
I have a couple of minor things that I have comments about.  Otherwise, it looks pretty good.  I'll test this out in a little bit; once I get the test done, I'll give an explicit approval.  Thanks @mikaelarguedas !

File: joy/CMakeLists.txt
Comment (mikaelarguedas) [reply joy/CMakeLists.txt:37]: For now this package doesn't have include directories and doesn't build any libraries so I think that we're good for now

File: joy/CMakeLists.txt
Comment (mikaelarguedas) [reply joy/CMakeLists.txt]: good point, fixed in https://github.com/ros2/joystick_drivers/pull/1/commits/143827fb7f4a33ffd90046afd6b1b06b398b6265

File: joy/src/joy_node_linux.cpp
Comment (clalancette) [inline joy/src/joy_node_linux.cpp:115]: I couldn't quite get things working with this node, with the robot jerking around instead of smoothly following the joystick commands.  After a bit of debugging, setting the autorepeat_rate to 0 is the reason why that is.  In the old version of the code I did, I had autorepeat_rate set to 20, which is how the ROS1 cartographer startup code configures it.  If I set this to 20 in this node, then things start working again, so I think that is what we should do here.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: joy/src/joy_node_linux.cpp
Comment (mikaelarguedas) [reply joy/src/joy_node_linux.cpp:115]: @clalancette  Done https://github.com/ros2/joystick_drivers/pull/1/commits/c78c12b243c00b8f6bd3e1f3190ebc084ffbbaef

**Review by clalancette — APPROVED** [review]
Excellent.  I'm happy with this now.  Thanks!

**Comment by mikaelarguedas** [comment]
Ok I squashed the commits and kept the renaming of the file in a separate commit to keep history.

Ready to merge this, the only pending question is what to do with changelogs and version number

**Comment by clalancette** [comment]
Oh, right, I forgot to comment about that.

For better or worse, on all of the other packages I've ported, I've kept the version number the same as the upstream one.  Since this is in a different "namespace", we shouldn't really have conflicts, and for most of these we aren't changing any core functionality.  Additionally, I'm not sure how we could consistently bump the version number and not run into a version number conflict with the upstream package anyway.  Thus, my vote is to keep the version number as-is.

As far as the changelog is concerned, I would personally leave it as-is as well.  While we aren't doing "releases" on any of this yet, the past version stuff still makes some sense, and could possibly be continued in the future.

**Comment by mikaelarguedas** [comment]
Sounds good to me, we can always revisit version numbers when we decide to release ROS2 1.0.

Merging then

**Pull request merged**
